### ISSUE TITLE
Warning about several instances of `styled-components` with Webpack 4 optimization.splitChunks

### ISSUE THREAD

**Issue by alfonsomunozpomer** [styled-components/styled-components#2170]
## System:
 - OS: macOS High Sierra 10.13.6
 - CPU: x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
 - Memory: 162.12 MB / 16.00 GB
 - Shell: 3.2.57 - /bin/bash
## Binaries:
 - Node: 8.6.0 - ~/.nvm/versions/node/v8.6.0/bin/node
 - Yarn: 1.2.1 - /usr/local/bin/yarn
 - npm: 6.4.1 - ~/.nvm/versions/node/v8.6.0/bin/npm
## npmPackages:
 - styled-components: ^4.0.2 => 4.0.2 

## Steps to reproduce
1. Create a package with Webpack 4 and two entry points that import `styled-components`.
2. Add `optimization.splitChunks` to `webpack.config.js` and create an explicit `vendors.bundle.js` for `node_modules.
3. Create bundles and verify that only a single instance of `styled-components` exist in `vendors.bundle.js`.
4. Create a HTML page that includes once `vendors.bundle.js`.

## Expected Behavior
The console shouldn’t display any warning about multiple instances of `styled-components` being initialized.

## Actual Behavior
The console displays the message `It looks like there are several instances of 'styled-components' initialized in this application...`.

**Comment by probablyup** [comment]
You need to make sure it's only in a single bundle. If you're getting that warning it's not being deduped properly. styled-components is a singleton library like react and loading multiple copies of it will cause all sorts of problems.

**Comment by alfonsomunozpomer** [comment]
I’ve created a repo where you can test the issue described above. To inspect the bundles:
```
git clone https://github.com/alfonsomunozpomer/styled-components-webpack-4.git
cd styled-components-webpack-4/
git checkout split-chunks
npm install
npx webpack --mode=development --profile --json > ./stats.json
npx webpack-bundle-analyzer ./stats.json
```

And then run the example:
```
npx webpack-dev-server --mode=development
```

And browse to `localhost:9000`.

**Comment by alfonsomunozpomer** [comment]
@probablyup I’ve been investigating a bit, since it was clear from my bundles that `styled-components` (and React, for that matter) was included in a single bundle but evaluated twice. [It turns out that this is a change in behaviour from Webpack 3](https://github.com/webpack/webpack/issues/7556). It would be a good idea to add a Webpack 4 configuration snippet to [the FAQ related section](https://www.styled-components.com/docs/faqs#why-am-i-getting-a-warning-about-several-instances-of-module-on-the-page). If this is hosted somewhere I can submit a PR.

**Comment by alfonsomunozpomer** [comment]
@probablyup [Found it](https://github.com/styled-components/styled-components-website/blob/master/sections/faqs/duplicated-styled-components.md).

**Comment by joetidee** [comment]
Could do with updating the FAQ page with a `splitChunks` example.

**Comment by alfonsomunozpomer** [comment]
@joetidee I did shortly after I reported the problem: https://github.com/styled-components/styled-components-website/pull/378

**Comment by joetidee** [comment]
@alfonsomunozpomer I can see an example for CommonsChunkPlugin, but not splitChunks.

**Comment by alfonsomunozpomer** [comment]
@joetidee Yes, the PR hasn’t been merged, here’s the proposed commit:
https://github.com/styled-components/styled-components-website/pull/378/commits/3f8bc6e1360f29b60ccb7e6c4f707b1fd069db27

**Comment by santteegt** [comment]
> @joetidee Yes, the PR hasn’t been merged, here’s the proposed commit:
> [styled-components/styled-components-website@3f8bc6e](https://github.com/styled-components/styled-components-website/commit/3f8bc6e1360f29b60ccb7e6c4f707b1fd069db27)

When I apply those optimization settings to my library & run it with `webpack --watch --mode=development`, it cannot be imported. When e.g. `import Lib from 'my library'` the value for Lib is either `undefined` or `1`. What could be the problem?

**Comment by alfonsomunozpomer** [comment]
@santteegt I’d be happy to help if there’s a repo I can clone to test.

**Comment by santteegt** [comment]
HI @alfonsomunozpomer 

Sure. Thanks for your help!!

Here's my [repo](https://github.com/santteegt/widget-example).

**How to reproduce**

To test how it works start following the instructions in the README. If you build the project with `npm run build` and then run the example app, it works without issues. But if you execute the project with `npm run dev` (webpack --watch --mode=development), the message `Warning about several instances of styles components...` appears. Then, if you enable the `optimization` setting on `webpack.config.js` the warning message disappears. However, the project library cannot be imported anymore

**Comment by alfonsomunozpomer** [comment]
A couple of comments:
- I think the reason why you don’t see the warning `It looks like there are several instances of 'styled-components' initialized in this application.` in production mode is because warning messages are stripped away in the production bundles, but it’s imported several times nonetheless.
- When the optimization is enabled, be aware that a second bundle, `vendors.js`, is generated in `lib`. Double check that you’re including it in your application. I’m not very familiar with TS projects but I think that’s the reason why it breaks.

**Comment by Robbiesmo** [comment]
Check out this bounty that pays out 45.0 DAI https://gitcoin.co/issue/santteegt/widget-example/1/3695  #TypeScript #JavaScript #webpack #styled-components #react

**Comment by Robbiesmo** [comment]
santteegt Fix codebase for a proper widget rendering #1

**Comment by Robbiesmo** [comment]
https://gist.github.com/Robbiesmo.com
